### ISSUE TITLE
Bug / Feature Request: Unquoted Numeric Primary Key Values from HTTP Server are Interpreted as Integar.

### ISSUE THREAD

**Issue by MrChadMWood** [pubkey/rxdb#6089]
BUG:

While using HTTP replication with Dixie storage, I've noticed that odd behaviors occur if the HTTP server provides an unquoted integer for the primaryKey value. A console error is thrown, stating that `primaryKey.trim() is not a function`. This somehow cascaded into other issues, and some unrelated React components began to act weird. I manually reviewed IndexedDB and found that the primary key values were in fact being stored as integers.

I don't think there's any actual standard on how to interpret integers versus strings in the query parameters, (REF: [RFC3986](https://datatracker.ietf.org/doc/html/rfc3986#section-3.4)), but it would be good IMO to try automatically casting the primaryKey value to string. I feel this way because RxDB seems to require `string` type for the primary key, so it's a bit strange to me that it wouldn't at-least try enforcing that via casting.

I resolved this issue by modifying my server component to quote the primary key value. Depending on perspective, I think this can be a bug or feature request.

**Comment by pubkey** [comment]
PrimaryKey must be a string.

**Comment by MrChadMWood** [comment]
That's a fair point. At the same time, why isn't an unquoted number interpreted as string if the requirement is that the value be a string? It would not need to be quoted if it contained any text, but if the value is all numbers then it will be assumed to be an integer even though that will cause strange side-effects. The behavior seems less intuitive to me.

At the very least, perhaps a better error would be a good idea. Something like, "Error: primary key is not a string". Otherwise, users may spend much spend time debugging why their app is acting strange because the current error is just `primaryKey.trim() is not a function`.

Not sure if I'm missing something though. Thanks for your attention, I love the product :)

**Comment by pubkey** [comment]
These transformations and checks would decrease performance. Just pass the correct data format into RxDB.
Dev-Mode will throw already if you do not define the primary key as string.

**Comment by MrChadMWood** [comment]
I am using dev-mode plugin. An error was thrown when I tried defining primary key as another type, so I changed it to string and there were no more errors. 

My API continued sending records as `{“primaryKey”:1234}`. RxDB seems to have stored the primary key as integer inside indexedDB, because it was unquoted (even with the schema defined primary key as string). As soon as I adjust the API to send `{“primaryKey”:”1234”}`, I no longer got the `primaryKey.trim() is not a function` error.

The solution is fine for me. I just modified the API to serialize the field into string type at response time. If I did not have such control over the backend, I’m not sure what the right approach would be. Maybe some modifications in the pull handler?
